src/setfit/integrations.py

```diff
@@ -0,0 +1,32 @@
+import importlib.util
+
+from transformers.trainer_utils import BestRun
+
+
+def is_optuna_available():
+    return importlib.util.find_spec("optuna") is not None
+
+
+def default_hp_search_backend():
+    if is_optuna_available():
+        return "optuna"
+
+
+def run_hp_search_optuna(trainer, n_trials: int, direction: str, **kwargs) -> BestRun:
+    import optuna
```

Comment:
Let's add a note to the source inspiration :)

```suggestion
    import optuna
    
    # Heavily inspired by transformers.integrations.run_hp_search_optuna
    # https://github.com/huggingface/transformers/blob/cbb8a37929c3860210f95c9ec99b8b84b8cf57a1/src/transformers/integrations.py#L160
```